### DISCUSSION TITLE
When using browser mode, is it possible to access the browser/context/page instance?

### DISCUSSION THREAD

**Discussion by awdyson** [vitest-dev/vitest#4474] [Q&A]
I'd like to be able to do things like change the viewport for a given test.

**Comment by sheremet-va** [comment]
No, there is currently no API to get access to those

**Reply by Omarov3000** [reply]
Do you plan to expose such api in future, or is it goes beyond the scope of vitest and we should use other tools?

**Reply by sheremet-va** [reply]
We are planning to expose a limited API, not the playwright/webdriverio page object. Closer to what Cypress exposes. But features will be limited to integration testing, nothing fancy - definitely no API interception and such.

**Comment by pfteter** [comment]
Can this be a feature requirement?

It should be possible to achieve this with custom commands, but that is hard to manage.

`page.evaluate()` would be really useful for Playwright in some special cases.

**Reply by sheremet-va** [reply]
Why do you need evaluate when you can just run the code? Test code runs in the browser already
